cap/modules/deposit/api.py

```diff
@@ -208,13 +208,25 @@ def upload(self, pid=None, *args, **kwargs):
         with UpdateDepositPermission(self).require(403):
             if request:
                 data = request.get_json()
-                _, record = request.view_args.get('pid_value').data
+
+                # retrieve the parameters and validate
+                data_url = data['url']
+                data_type = data['type']
+                download = data['for_download']
```

Comment:
nit: maybe rename params in data the same way? download instead of for_download, webhook instead of for_connection?

Reply:
url instead of data_url
type_ instead of data_type